framework/src/play/cache/Cache.java

```diff
@@ -63,17 +66,25 @@ public static void clear() {
 
   /** Initialize the cache system. */
   public static void init() {
-    if ("enabled".equals(Play.configuration.getProperty("memcached", "disabled"))) {
-      try {
-        cacheImpl = new MemcachedImpl(Play.configuration);
-        logger.info("Connected to memcached");
-      } catch (Exception e) {
-        logger.error("Error while connecting to memcached", e);
-        logger.warn("Fallback to local cache");
-        cacheImpl = EhCacheImpl.newInstance();
-      }
-    } else {
-      cacheImpl = EhCacheImpl.newInstance();
+    CacheImpl cache;
+    try {
+      Class<?> klass = Class.forName("play.cache.CacheImpl");
+      Constructor<?> constructor = klass.getDeclaredConstructor();
+      constructor.setAccessible(true);
+      cache = (CacheImpl) constructor.newInstance();
+    } catch (ClassNotFoundException
+             | NoSuchMethodException
+             | InvocationTargetException
+             | InstantiationException
+             | IllegalAccessException e) {
+      cache = new DummyCacheImpl();
```

Comment:
```suggestion
      cache = new DummyCacheImpl();
      logger.info("Using dummy cache (no caching)");
```

Reply:
I prefer not to. It's already waaaaaaaay too chatty at startup for my taste. I know there's no caching "in RePlay" as I did not set it up. If I need it I'll learn about it.

But we have a ton of caching still in many other places: hibernate, redis, jvm, CPU... It's not very clear what this "no caching" means. So I rather do without.

Reply:
@cies The problem is when I **want** to use cache, and I add "ehcache" or "memcached" module. BUT `ClassNotFoundException` happens. Then I definitely need to see some error in log to understand what was the problem.

Reply:
@asolntsev Sorry to've bothered you with the review: I had some insights and found that the code I submitted for review was bad in so many aspects. I've just pushed a new attempt, that fixes all feedback, uses my new understandings and adds tests.